WordPressAuthenticator/Authenticator/WordPressAuthenticator.swift

```diff
@@ -309,68 +305,91 @@ import AuthenticationServices
 
     // MARK: - Authentication Link Helpers
 
-
     /// Present a signin view controller to handle an authentication link.
     ///
     /// - Parameters:
     ///     - url: The authentication URL
-    ///     - rootViewController: The view controller to act as the presenter for the signin view controller.
-    ///                           By convention this is the app's root vc.
+    ///     - rootViewController: The view controller to act as the presenter for the signin view controller.  By convention this is the app's root vc.
+    ///     - automatedTesting: for calling this method for automated testing.  It won't sync the account or load any other VCs.
```

Comment:
I wonder if there's another way to call this method in unit tests without using a flag? Perhaps we can inject a `Storyboard` instance, which could be nil in case of testing? Not sure if it's possible in this case, but the flag might be harder to maintain and could lead to bugs that are not immediately visible.

Reply:
I agree this isn't ideal, but it was the best solution in terms of simplicity and scope-tightness that I could find.

In my modest opinion the reason I'm having to do this is because the code isn't well architected.  I could try to inject the code to "continue" but the problem is that this method exists at the boundary between WPiOS and WPAuth... meaning that the logic for spawning the VCs is defined now within WPAuth, and if I made that code injectable, it would need to be moved to WPiOS.

I don't know if that makes sense, but basically injecting logic would mean moving that logic to all host Apps (WCiOS, WPiOS, Woo, etc).

As in the reply below though - I don't particularly disagree... the reason I chose the boolean instead is to cause the smallest possible impact as part of a bugfix (since these PRs are not really enhancements).

Reply:
I see. If the price is to have an impact on all the apps, then we should probably leave as it is for now.